core/src/processing/core/PGraphics.java

```diff
@@ -7996,7 +7996,25 @@ public final float brightness(int rgb) {
   public int lerpColor(int c1, int c2, float amt) {  // ignore
     return lerpColor(c1, c2, amt, colorMode);
   }
-
+  
+  
+   /**
+   * Generalization of the function lerpColor for many colors that are passed 
+   * by parameter as a vector of colors
+   *
+   * @webref color:creating_reading
+   * @usage web_application
+   * @param c[] interpolate to this set of color
+   * @param amt between 0.0 and 1.0
+   * @see PImage#blendColor(int, int, int)
+   * @see PGraphics#color(float, float, float, float)
+   * @see PApplet#lerp(float, float, float)
+   */
+  int lerpColor(int c[], float amt) {
+    int index = int(c.length * amt);
```

Comment:
That's a syntax error outside of the PDE. You meant `(int)(c.length * amt)`. Also, index is now (c.length - 1) for all of the final (e.g.) ⅓ of a three-color gradient, giving the result shown:
![wrong](https://cloud.githubusercontent.com/assets/5357642/26529449/deba9b78-43b7-11e7-9818-08d06bfa4239.png)
So what you really need is
```java
int index = (int)((c.length-1) * amt);
```
Then you need to check the boundaries with if blocks so if index is not out of range (i.e. `index < 0` or `index > c.length-1`) you set it to within the range.